lib/rules/aria-dialog-name.json

```
@@ -0,0 +1,13 @@
+{
+	"id": "aria-dialog-name",
+	"selector": "[role=\"dialog\"], [role=\"alertdialog\"]",
+	"matches": "no-naming-method-matches",
+	"tags": ["cat.aria", "wcag2a", "wcag412"],
```

Looking good, glad all the tests are passing now. Just one more thing before we can merge. The rule should be a best practice rule and not linked to wcag success criterion

```suggestion
	"tags": ["cat.aria", "best-practice"],
```

---

Right, I've been meaning to ask how this is decided. @WilcoFiers added tags in the umbrella issue but some of the newly implemented rules don't match those tags (or just partially):
- aria-command-name has "cat.aria", "wcag2a", "wcag412" instead of just "wcag2a"
- aria-progressbar-name has "cat.aria", "wcag2a", "wcag111" instead of just "sc111"
- aria-tooltip-name has "cat.aria", "wcag2a", "wcag412" instead of "best-practice"

Specifically here: Why is not naming a dialog not a 4.12 failure and "just" best practice?

---

This is based on our interpretation of WCAG 2. Every rule needs a category tag. That's the "cat.*" stuff. Additionally, a rule is either a WCAG success criterion, which are those "wcag123" type tags or a "best-practice". If it's a wcag tag, it'll also include a level, which is just the level of that success criterion (SC) in WCAG. So SC 1.1.1 has the tag "cat.aria", "wcag2a", "wcag111". This best practice would get "cat.aria", "best-practice".

As for why I made tooltip a WCAG rule and think dialog should be a best practice, that is based on some discussions I've had internally at Deque. A tooltip doesn't make much sense when it is empty, but more importantly a tooltip is a piece of a user interface that users will understand to be a single "thing". Success criterion 4.1.2 applies only to "user interface components". It's a little more difficult to argue that a dialog is a user interface component, since it is just a region of the screen with other content in it. User interface components are the smallest things users will understand as its own control. For example a select is a user interface component, but a list of radio buttons is not, each radio button would be though.

---

I can see how this makes sense. 

Slightly going off-topic here: why does a `tree` not fall under "user interface components"? It was was scratched due to `aria-structure-name` being scratched but I would consider this more of a widget and not just some region. Happy to continue this in https://github.com/dequelabs/axe-core/issues/2421#issuecomment-717875355